app/src/main/java/org/mozilla/fenix/gleanplumb/Message.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.gleanplumb
+
+import org.mozilla.fenix.nimbus.MessageData
+import org.mozilla.fenix.nimbus.StyleData
+
+/**
+ * A data class that holds a representation of GleanPlum message from Nimbus.
+ *
+ * @param id identifies a message as unique.
+ * @param data Data information provided from Nimbus.
+ * @param action A strings that represents which action should be performed
+ * after a message is clicked.
+ * @param style Indicates how a message should be styled.
+ * @param triggers A list of strings corresponding to targeting expressions. The message
+ * will be shown if all expressions `true`.
+ * @param metadata Metadata that help to identify if a message should shown.
+ */
+data class Message(
+    val id: String,
+    val data: MessageData,
+    val action: String,
+    val style: StyleData,
+    val triggers: List<String>,
+    val metadata: Metadata
+) {
+    /**
+     * A data class that holds metadata that help to identify if a message should shown.
+     *
+     * @param id identifies a message as unique.
+     * @param displayCount Indicates how many times a message is displayed.
+     * @param pressed Indicates if a message has been clicked.
+     * @param dismissed Indicates if a message has been closed.
+     */
+    data class Metadata(
+        val id: String,
+        val displayCount: Int = 0,
```

Comment:
Does this default make sense for Nimbus? I'm honestly not sure myself. Maybe it should be 1?

No default might also be less error-prone if you're also unsure.

Reply:
For the moment, the messages metadata is only been tracked on the apps to identify when a message should be shown of not, Nimbus only indicates the maximum times that a messages should be shown. We were using a `displayCount` of `0` to signal that the message was not display at that moment, as if we are initializing it with one ,it could indicate that a message is shown right away after creation, a message could be created without being shown, as could be provide by Nimbus, but not shown right away until all conditions are met :) .